pkg/kubelet/kubelet.go

```diff
@@ -1150,6 +1152,18 @@ func (kl *Kubelet) GetBoundPods() ([]api.BoundPod, error) {
 	return kl.pods, nil
 }
 
+// GetPodFullName provides the first pod that matches namespace and name, or false
+// if no such pod can be found.
+func (kl *Kubelet) GetPodFullName(namespace, name string) (string, bool) {
```

Comment:
Doing this makes me realize how podFullName as a concept needs to be nuked and replaced with something better.  Usually you're looking up something by a series of qualifiers - namespace / name / attempt uuid.  This change at least steers us a bit closer to that though.

----- Original Message -----
> > @@ -1150,6 +1152,18 @@ func (kl *Kubelet) GetBoundPods() ([]api.BoundPod,
> > error) {
> >  	return kl.pods, nil
> >  }
> >  
> > +// GetPodFullName provides the first pod that matches namespace and name,
> > or false
> > +// if no such pod can be found.
> > +func (kl *Kubelet) GetPodFullName(namespace, name string) (string, bool) {
> 
> We already have a free function of the same name.  Can this be something like
> GetPodByName(), return *BoundPod, and let the caller call GetPodFullName() ?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3270/files#r22600841